pkg/registry/registry.go

```diff
@@ -59,32 +60,61 @@ type registry struct {
 	log        logr.Logger
 }
 
-func (r *registry) writeImageRegistryCredentials(ctx context.Context) error {
-	_, err := r.kubeClient.GetNamespace(ctx, pullSecretNamespace, metav1.GetOptions{})
-	if err != nil {
-		r.log.Info("Can not find namespace for pull secrets, assuming vanilla k8s")
-		return nil
+type dockerAuth struct {
+	Auth  string
+	Email string
+}
+
+func (r *registry) registryFromImageURL(image string) (string, error) {
+	u, err := url.Parse(image)
+	if err != nil || u.Host == "" {
+		// "reg.io/org/repo:tag" are invalid from URL scheme POV
+		// prepending double slash in case of error or empty Host and trying to parse again
+		u, err = url.Parse("//" + image)
+	}
+
+	if err != nil || u.Host == "" {
+		return "", fmt.Errorf("failed to parse image url: %s", image)
 	}
 
+	return u.Host, nil
```

Comment:
WDYT of this? Doesn't require the "//" hack. I'm guessing your `//another-registry.io/org/repo@sha256:987` test case fails then, but is it legitimate?
```suggestion
	if parts := strings.SplitN(c, "/", 2); len(parts) >= 2 {
		return parts[0], nil
	} else {
		return "", errors.New("could not determine registry: invalid host")
	}
```

Reply:
I actually wanted to avoid string splitting and use proper methods from std lib.
If you'd like I can accept your suggestion (note: this is already merged downstream)

> is it legitimate?

Depends on point of view. I haven't seen image references that would start with `//`. But on the other hand it's incorrect from URL/URI perspective.

Reply:
I think they can start with `oci://`, though since I cannot find any good reference document on container image naming, I would stick to your code.